uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/calender-instance_test.js

```diff
@@ -0,0 +1,91 @@
+define([
+  'mocks/data/work-pattern-data',
+  'leave-absences/shared/models/instances/calender-instance',
+], function (mockData) {
+  'use strict';
+
+  describe('CalenderInstance', function () {
+    var CalenderInstance;
+
+    beforeEach(module('leave-absences.models.instances'));
+
+    beforeEach(inject([
+      "CalenderInstance",
+      function (_CalenderInstance_) {
+        CalenderInstance = _CalenderInstance_;
+        CalenderInstance.calenderData = mockData.calenderData().values;
+      }]
+    ));
+
+    describe('isWorkingDay()', function () {
+
+      var dateToSearch;
+
+      it('return true if it is a working day', function () {
+        dateToSearch = mockData.calenderData().values[0].date;
```

Comment:
What if the mocked data gets changed and `values[0]` is not a working day anymore? Instead of picking an arbitrary index, use `.find()` to find for sure a working day

(same applies for all the other tests for the `isXYZ()` methods)